### PR TITLE
Fix torch.tensor call with tuple in Dynamo

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114383

Partially fixes https://github.com/pytorch/pytorch/issues/114231

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/_dynamo/variables/torch.py [hunk 1]
```diff
@@ -575,7 +575,7 @@ def check_any_unspec(x):
                     # NB: This includes UnspecializedPythonVariable
                     if isinstance(x, (TensorVariable, SymNodeVariable)):
                         return True
-                    elif isinstance(x, ListVariable):
+                    elif isinstance(x, (ListVariable, TupleVariable)):
```
Comment (voznesenskym) [inline torch/_dynamo/variables/torch.py:578]: the dynamo pattern here is unpack_var_sequence check and access, should work for all collections.

File: torch/_dynamo/variables/torch.py [hunk 2]
```diff
@@ -575,7 +575,7 @@ def check_any_unspec(x):
...
                         return any(check_any_unspec(y) for y in x.items)
                     # TODO: there maybe other recursive structures you need to
                     # check
```
Comment (voznesenskym) [inline torch/_dynamo/variables/torch.py:581]: reeeeee this whole func needs a good hard revisit

**Comment by ezyang** [comment]
looks like someone merged something equivalent

**Pull request closed without merging**